### PR TITLE
TINY-7602: Fix throbber focus logic not working with pastebins

### PR DESCRIPTION
**Pull request by ltrouton**
Related Ticket: TINY-7602, TINY-7373

Description of Changes:
* Update the throbber focus code (TINY-7373) to not steal focus from the pastebin when pasting in IE11

Pre-checks:
* [X] Changelog entry added (Haven't added a changelog since the code causing the issue hasn't be released)
* [X] Tests have been added (if applicable)
* [X] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [X] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/themes/silver/main/ts/ui/throbber/Throbber.ts [hunk 1]
```diff
@@ -103,12 +106,19 @@ const renderThrobber = (spec): AlloySpec => ({
   components: [ ]
 });
 
+const isPasteBinTarget = (target: unknown) =>
+  Optional.from(target)
+    .map(SugarElement.fromDom)
```
Comment (lnewson) [inline modules/tinymce/src/themes/silver/main/ts/ui/throbber/Throbber.ts:111]: This isn't going to work nicely when the target is an editor. Basically, we can't be converting to a SugarElement, without first knowing we're dealing with a `Node`.

File: modules/tinymce/src/themes/silver/test/ts/browser/throbber/ThrobberFocusTest.ts [hunk 2]
```diff
@@ -189,4 +191,30 @@ describe('browser.tinymce.themes.silver.throbber.ThrobberFocusTest', () => {
...
+    const normalDiv = SugarElement.fromHtml<HTMLElement>('<div class="not-pastebin" >&nbsp;</div>');
+    const fakePasteBin = SugarElement.fromHtml<HTMLElement>('<div class="mce-pastebin">&nbsp;</div>');
+
+    Insert.append(editorBody, normalDiv);
+    Insert.append(editorBody, fakePasteBin);
+    await pEnableThrobber(editor, pAssertThrobberFocus);
+    FocusTools.setFocus(SugarBody.body(), '#tempInput');
+
+    // Make sure throbber doesn't take focus when the pastebin is the target
+    editor.dom.fire(fakePasteBin.dom, 'focusin');
```
Comment (lnewson) [inline modules/tinymce/src/themes/silver/test/ts/browser/throbber/ThrobberFocusTest.ts:207]: Can we please not use fake events here and instead actually focus the element using `Focus.focus(fakePasteBin)`?

Comment (ltrouton) [reply modules/tinymce/src/themes/silver/test/ts/browser/throbber/ThrobberFocusTest.ts:207]: I had tried that but was unable to get it to trigger the listeners set up in Throbber.ts

Comment (ltrouton) [reply modules/tinymce/src/themes/silver/test/ts/browser/throbber/ThrobberFocusTest.ts:207]: I thought I had tried this before, but adding `tabindex` to the divs seems to allow `Focus.focus` to be used.

Comment (lnewson) [reply modules/tinymce/src/themes/silver/test/ts/browser/throbber/ThrobberFocusTest.ts:207]: The problem isn't really that from what I can see, it's that we're not setting up the pastebin correctly. If you look at https://github.com/tinymce/tinymce/blob/develop/modules/tinymce/src/plugins/paste/main/ts/core/PasteBin.ts#L40 you'll see we add a number of other attributes (the contenteditable one specifically). To properly test this, we should be setting up a more realistic paste bin.

Comment (lnewson) [reply modules/tinymce/src/themes/silver/test/ts/browser/throbber/ThrobberFocusTest.ts:207]: In this case, it's likely the `contenteditable="true"` attribute that's needed to handle the focus call.

File: modules/tinymce/src/themes/silver/main/ts/ui/throbber/Throbber.ts (hunk 1 above)
Comment (lnewson) [reply modules/tinymce/src/themes/silver/main/ts/ui/throbber/Throbber.ts:111]: To clarify more, the type for `target` here should be `Editor | Node`

File: modules/tinymce/src/themes/silver/test/ts/browser/throbber/ThrobberFocusTest.ts (hunk 2 above)
Comment (ltrouton) [reply modules/tinymce/src/themes/silver/test/ts/browser/throbber/ThrobberFocusTest.ts:214]: Have added a note as to why we can't use `Focus.focus` here

File: modules/tinymce/src/themes/silver/main/ts/ui/throbber/Throbber.ts (hunk 1 above)
Comment (ltrouton) [reply modules/tinymce/src/themes/silver/main/ts/ui/throbber/Throbber.ts]: Thanks for pointing this out. This has been reworked now with your provided snippets

**Review by lnewson — APPROVED** [review]

**Review by ashfordneil — APPROVED** [review]

**Review by metricjs — APPROVED** [review]

**Pull request merged**